kong/pdk/service/response.lua

```diff
@@ -285,10 +285,11 @@ local function new(pdk, major_version)
   -- -- or `access` phase prior calling this function.
   --
   -- local body = kong.service.response.get_raw_body()
-  function response.get_raw_body()
+  function response.get_raw_body(buffered_proxying)
     check_phase(header_body_log)
     local ctx = ngx.ctx
-    if not ctx.buffered_proxying then
+    local buffered = buffered_proxying or ctx.buffered_proxying
```

Comment:
I am not sure about that. Adding `buffered_proxying` as public interface parameter feels a bit wrong. Also what if someone passes `false` here?

Reply:
Yes this seems a little be confused if users pass in `false` but we will still respect the `ctx.bufferred_proxying`.
My intention was to reduce the read of `ctx`, because local variable is faster than `ctx`, and when invoke `get_raw_body` inside `get_body` I could just pass in the `ctx.bufferd_proxying` we've read already. 
I think we can revert the `get_raw_body` change, and inside the `get_body` function, just replace the `get_raw_body` call with `return ctx.buffered_body or ""`, because we already do the `chech_phase` and `ctx.buffered_proxying` check, what do you think?

Reply:
> I am not sure about that. Adding `buffered_proxying` as public interface parameter feels a bit wrong. Also what if someone passes `false` here?

Yes this seems a little be confused if users pass in false but we will still respect the ctx.bufferred_proxying.
My intention was to reduce the read of ctx, because local variable is faster than ctx, and when invoke get_raw_body inside get_body I could just pass in the ctx.bufferd_proxying we've read already.
I think we can revert the get_raw_body change, and inside the get_body function, just replace the get_raw_body call with return ctx.buffered_body or "", because we already do the chech_phase and ctx.buffered_proxying check, what do you think?

Reply:
Yes, I would revert these PDK API changes (where we add new parameters). Not worth it, and paints us in a corner.